v2/internal/controllers/containerservice_managedcluster_v1api20230701_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	aks "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230701"
+	"github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_AKS_ManagedCluster_20230701_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	adminUsername := "adminUser"
+	sshPublicKey, err := tc.GenerateSSHKey(2048)
+	tc.Expect(err).ToNot(HaveOccurred())
+
+	cluster := NewManagedCluster20230701(tc, rg, adminUsername, sshPublicKey)
+
+	tc.CreateResourceAndWait(cluster)
+
+	tc.Expect(cluster.Status.Id).ToNot(BeNil())
+	armId := *cluster.Status.Id
+
+	// Perform a simple patch
+	skuName := aks.ManagedClusterSKU_Name_Base
+	skuTier := aks.ManagedClusterSKU_Tier_Standard
+	old := cluster.DeepCopy()
+	cluster.Spec.Sku = &aks.ManagedClusterSKU{
+		Name: &skuName,
+		Tier: &skuTier,
+	}
+	tc.PatchResourceAndWait(old, cluster)
+	tc.Expect(cluster.Status.Sku).ToNot(BeNil())
+	tc.Expect(cluster.Status.Sku.Name).ToNot(BeNil())
+	tc.Expect(*cluster.Status.Sku.Name).To(Equal(aks.ManagedClusterSKU_Name_STATUS_Base))
+	tc.Expect(cluster.Status.Sku.Tier).ToNot(BeNil())
+	tc.Expect(*cluster.Status.Sku.Tier).To(Equal(aks.ManagedClusterSKU_Tier_STATUS_Standard))
+
+	// Run sub tests
+	tc.RunSubtests(
+		testcommon.Subtest{
+			Name: "AKS KubeConfig secret & configmap CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				AKS_ManagedCluster_Kubeconfig_20230701_OperatorSpec(tc, cluster)
+			},
+		})
+	tc.RunParallelSubtests(
+		testcommon.Subtest{
+			Name: "AKS AgentPool CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				AKS_ManagedCluster_AgentPool_20230701_CRUD(tc, cluster)
+			},
+		},
+	)
+
+	tc.DeleteResourceAndWait(cluster)
+
+	// Ensure that the cluster was really deleted in Azure
+	exists, retryAfter, err := tc.AzureClient.CheckExistenceWithGetByID(tc.Ctx, armId, string(aks.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func NewManagedCluster20230701(tc *testcommon.KubePerTestContext, rg *v1api20200601.ResourceGroup, adminUsername string, sshPublicKey *string) *aks.ManagedCluster {
+	region := to.Ptr("westus3") // TODO: the default test region of westus2 doesn't allow ds2_v2 at the moment
```

Comment:
can move this to Location with comment?

Reply:
and remove `//region := tc.AzureRegion`

Reply:
When a test can't be run in the standard test region, this is how we normally move it, so that every resource gets the same region (if there are more than one).